enclave/core/ctype.c

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <ctype.h>
+
+static const unsigned short _ctype_b[384] = {
```

Comment:
Do we need ctype in core? If so can we just use MUSL's implementation?

Reply:
This is used by oe_strtoul(), which is used by oe_printf().

Reply:
It's not clear that we need all of ctype as implemented but it is a dependency for other pieces of the elibc implementations (printf, strtoul, string … see other comments)

We cannot just use MUSL's implementation because it (the body of ctype source files) pulls in significant amount of dependencies, most notably locale_impl that does not play well in enclave core and ends up introducing potential dupe symbols like C_LOCALE when linking against oelibc. 

As a bit of background, the elibc implementation originated from determining the subset of C library functionality sufficient for an enclave internal library (mbedtls) without having to build oelibc, where much of the MUSL source code could not compile on Windows. In some cases, I have directly compiled MUSL sources into oecore where it was clear there are few dependencies, but part of the original challenge that spawned the elibc was that many basic things (like ctype) end up being widely integrated with MUSL implementation headers that make them difficult to partition from the subsets of functionality we want in the enclave.